src/Orleans/Logging/LogManager.cs

```diff
@@ -0,0 +1,494 @@
+                    {
+                        TelemetryConsumers.Add(new TraceTelemetryConsumer());
+                    }
+                }
+
+                IsInitialized = true;
+            }
+        }
+
+        /// <summary>
+        /// Uninitialize the Orleans Logger subsystem in this process / app domain.
+        /// </summary>
+        public static void UnInitialize()
+        {
+            lock (lockable)
+            {
+                Close();
+                LogConsumers = new ConcurrentBag<ILogConsumer>();
+                TelemetryConsumers = new ConcurrentBag<ITelemetryConsumer>();
+
+                loggerStoreInternCache?.StopAndClear();
+
+                BulkMessageInterval = defaultBulkMessageInterval;
+                BulkMessageLimit = Constants.DEFAULT_LOGGER_BULK_MESSAGE_LIMIT;
+                IsInitialized = false;
+            }
+        }
+
+        internal static Severity GetDefaultSeverityForLog(string source, LoggerType logType)
+        {
+            string expandedName = logType + "." + source;
+
+            lock (lockable)
+            {
+                if (traceLevelOverrides.Count > 0)
+                {
+                    foreach (var o in traceLevelOverrides)
+                    {
+                        if (expandedName.StartsWith(o.Item1))
+                        {
+                            return o.Item2;
+                        }
+                    }
+                }
+            }
+
+            return logType == LoggerType.Runtime ? runtimeTraceLevel : appTraceLevel;
+        }
+
+        /// <summary>
+        /// Find the Logger with the specified name
+        /// </summary>
+        /// <param name="loggerName">Name of the Logger to find</param>
+        /// <returns>Logger associated with the specified name</returns>
+        internal static LoggerImpl FindLogger(string loggerName)
+        {
+            if (loggerStoreInternCache == null) return null;
+
+            LoggerImpl logger;
+            loggerStoreInternCache.TryFind(loggerName, out logger);
+            return logger;
+        }
+
+        /// <summary>
+        /// Find existing or create new Logger with the specified name
+        /// </summary>
+        /// <param name="loggerName">Name of the Logger to find</param>
+        /// <param name="logType">Type of Logger, if it needs to be created</param>
+        /// <returns>Logger associated with the specified name</returns>
+        internal static LoggerImpl GetLogger(string loggerName, LoggerType logType)
+        {
+            return loggerStoreInternCache != null ?
+                loggerStoreInternCache.FindOrCreate(loggerName, () => new LoggerImpl(loggerName, logType)) :
+                new LoggerImpl(loggerName, logType);
+        }
+
+        internal static LoggerImpl GetLogger(string loggerName)
+        {
+            return GetLogger(loggerName, LoggerType.Runtime);
+        }
```

Comment:
Could not replace all instances of LoggerImpl with logger, so these need to remains as in for now. :/